### DISCUSSION TITLE
Improvements to Find and Replace Utility

### DISCUSSION THREAD

**Discussion by rishifter** [craftcms/cms#15669] [Ideas]
We believe the Find and Replace utility provided in Craft could benefit from some improvements, which will come in handy especially for sites with a lot of content. It'd be nice if some of these new options are considered to be added to the utility:

1. Partial-word/Full-word search option
2. Dry run with option to selectively replace items ([similar](https://github.com/craftcms/cms/issues/4767))
3. Case-sensitive searches
4. Improved results when performing a replacement — how many items were replaced?

**Comment by TheFunkyMonk** [comment]
Would be nice to have some kind of an interstitial "review" screen too, where you could see a diff before committing to the changes.

**Comment by rishifter** [comment]
Bumping this for some attention please.

**Comment by foxtrotcz** [comment]
I guess preview would be nice. I would rather not destroy my site with badly defined search.

**Comment by sandrodunkel** [comment]
If we're already rethinking the find/replace utility: having the option to only run it on certain site would be helpful as well.
